subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c

```diff
@@ -1326,8 +1344,83 @@ uint32_t radio_tmr_start(uint8_t trx, uint32_t ticks_start, uint32_t remainder)
 
 	nrf_timer_cc_set(EVENT_TIMER, 0, remainder);
 
+#if defined(CONFIG_BT_CTLR_NRF_GRTC)
+	uint32_t cntr_l, cntr_h, cntr_h_overflow, stale;
+
+	/* Disable capture/compare */
+	nrf_grtc_sys_counter_compare_event_disable(NRF_GRTC, HAL_CNTR_GRTC_CC_IDX_RADIO);
+
+	/* Publish GRTC compare */
+	NRF_GRTC->PUBLISH_COMPARE[HAL_CNTR_GRTC_CC_IDX_RADIO] =
+		((HAL_EVENT_TIMER_START_PPI <<
+		  GRTC_PUBLISH_COMPARE_CHIDX_Pos) &
+		 GRTC_PUBLISH_COMPARE_CHIDX_Msk) |
+		((GRTC_PUBLISH_COMPARE_EN_Enabled <<
+		  GRTC_PUBLISH_COMPARE_EN_Pos) &
+		 GRTC_PUBLISH_COMPARE_EN_Msk);
+
+	/* Enable same DPPI in Global domain */
+	NRF_DPPIC20->CHENSET = BIT(HAL_EVENT_TIMER_START_PPI);
+
+	/* Setup PPIB send subscribe */
+	NRF_PPIB21->SUBSCRIBE_SEND[HAL_EVENT_TIMER_START_PPI] =
+		BIT(HAL_EVENT_TIMER_START_PPI) | PPIB_SUBSCRIBE_SEND_EN_Msk;
+
+	/* Setup PPIB receive publish */
+	NRF_PPIB11->PUBLISH_RECEIVE[HAL_EVENT_TIMER_START_PPI] =
+		BIT(HAL_EVENT_TIMER_START_PPI) | PPIB_PUBLISH_RECEIVE_EN_Msk;
+
+	/* NOTE: We are going to use TASKS_CAPTURE to read current
```

Comment:
(same)
Anyhow, shouldn't this logic should be refactored into a common function?